generator/tests/test_other_library/run_other_library_tests.py

```diff
@@ -20,231 +20,289 @@
+    gv.populate_error_list(ob)
     ValidationFiles.ValidationFiles(ob, True)
-    os.chdir('./temp')
-    newdir = global_variables.language
-    if not os.path.isdir(newdir):
-        os.mkdir(newdir)
-    os.chdir(newdir)
+    go_into_new_directory()
+
+    prefix = gv.prefix
     base_files = BaseClassFiles.BaseClassFiles(prefix,  ob['baseElements'], True)
     base_files.write_files()
     os.chdir('../../.')
 
+
 def generate_common_templates(filename):
-    parser = ParseXML.ParseXML(filename)
-    ob = parser.parse_deviser_xml()
-    for wc in ob['baseElements']:
-        strFunctions.prefix_classes(wc)
-    for working_class in ob['baseElements']:
-        if working_class['name'] == global_variables.document_class:
-            working_class['document'] = True
-    global_variables.populate_error_list(ob)
-    prefix = global_variables.prefix
-    os.chdir('./temp')
-    newdir = global_variables.language
-    if not os.path.isdir(newdir):
-        os.mkdir(newdir)
-    os.chdir(newdir)
+    """
+    Write the 'common' files. TODO need more explanation.
+
+    :param filename: the XML file to parse.
+    :return: nothing
+    """
+    ob = common_set_up(filename)
+    gv.populate_error_list(ob)
+    go_into_new_directory()
+
+    prefix = gv.prefix
     base_files = BaseClassFiles.BaseClassFiles(prefix,  ob['baseElements'], True)
     base_files.write_common_files()
     os.chdir('../../.')
 
+
 def generate_forward(filename):
-    parser = ParseXML.ParseXML(filename)
-    ob = parser.parse_deviser_xml()
-    for wc in ob['baseElements']:
-        strFunctions.prefix_classes(wc)
-    for working_class in ob['baseElements']:
-        if working_class['name'] == global_variables.document_class:
-            working_class['document'] = True
-    global_variables.populate_error_list(ob)
-    os.chdir('./temp')
-    newdir = global_variables.language
-    if not os.path.isdir(newdir):
-        os.mkdir(newdir)
-    os.chdir(newdir)
+    """
+    Write extension file with forward declaration of all opaque C types.
+
+    :param filename: the XML file to parse.
+    :return: nothing
+    """
+    ob = common_set_up(filename)
+    gv.populate_error_list(ob)
+    go_into_new_directory()
+
     ext = ExtensionFiles.ExtensionFiles(ob, 'fwd', True)
     ext.write_files()
     os.chdir('../../.')
 
+
 def generate_enum(filename):
-    parser = ParseXML.ParseXML(filename)
-    ob = parser.parse_deviser_xml()
-    for wc in ob['baseElements']:
-        strFunctions.prefix_classes(wc)
-    for working_class in ob['baseElements']:
-        if working_class['name'] == global_variables.document_class:
-            working_class['document'] = True
-    global_variables.populate_error_list(ob)
-    os.chdir('./temp')
-    newdir = global_variables.language
-    if not os.path.isdir(newdir):
-        os.mkdir(newdir)
-    os.chdir(newdir)
+    """
+    According to ExtensionFile.py, this writes the extension types file.
+    TODO what does that mean?
```

Comment:
For SBML the extension file includes all the enums that are associated with a package. When generting code for a different library we still need a file that lists any enums so we create a version of the same file. 

Need to review this issue #44

Reply:
docstring updated